### PR TITLE
Add environment variables for system, docker, and user

### PR DESCRIPTION
**Pull request by nicklathe**
- Variables default to empty strings in terraform *.tf files
    - Utilize existing templates to wrtie to config.yaml files

### CODE REVIEW

File: kubernetes/cloud-configs/controller.yaml [hunk 1]
```diff
@@ -660,4 +660,23 @@ write_files:
...
\ No newline at end of file
+      }
+  - path: /etc/systemd/system.conf.d/10-default-env.conf
+    filesystem: root
+    mode: 0644
+    contents:
+      inline: |
+        [Manager]
+        ${system_environment}
+  - path: /etc/docker-environment.env
```
Comment (weitenghuang) [inline kubernetes/cloud-configs/controller.yaml]: where do we use `/etc/docker-environment.env` and `/etc/profile.env`?

File: etcd/cloud-configs/etcd.yaml [hunk 2]
```diff
@@ -142,9 +142,29 @@ write_files:
...
+      inline: |
+        [Manager]
+        ${system_environment}
+  - path: /etc/docker-environment.env
+    filesystem: root
+    mode: 0644
+    contents:
+      inline: |
+        ${docker_environment}
+  - path: /etc/profile.env
```
Comment (weitenghuang) [inline etcd/cloud-configs/etcd.yaml]: is `/etc/profile.env` being used?

File: kubernetes/cloud-configs/controller.yaml (hunk 1 above)
Comment (nicklathe) [reply kubernetes/cloud-configs/controller.yaml]: I answered Wei's questions offline.

File: etcd/cloud-configs/etcd.yaml (hunk 2 above)
Comment (nicklathe) [reply etcd/cloud-configs/etcd.yaml]: Questions answered offline.

**Comment by nicklathe** [comment]
There are a 3 other `docker run` in the kubernetes s3-cloudconfig-bootstrap.sh file, but they are for retrieving etcd instances. Since the calls are internal only, I don't believe I need to feed in the docker environment variables into these containers. Let me know if that assumption is incorrect.

File: kubernetes/controller.tf [hunk 3]
```diff
@@ -44,6 +44,21 @@ variable "controller_encrypt_docker_volume" {
...
+variable "docker_environment" {
+  description = "Environment variables to be used by Docker."
+  default     = ""
+}
+
+variable "user_environment" {
+  description = "Environment variables to be used by the user."
+  default     = ""
+}
+
```
Comment (dragan) [inline kubernetes/controller.tf]: We only need to declare the variables once, move them to main.tf.

File: kubernetes/node.tf [hunk 4]
```diff
@@ -58,6 +58,21 @@ variable "node_encrypt_logging_volume" {
...
+variable "docker_environment" {
+  description = "Environment variables to be used by Docker."
+  default     = ""
+}
+
+variable "user_environment" {
+  description = "Environment variables to be used by the user."
+  default     = ""
+}
+
```
Comment (dragan) [inline kubernetes/node.tf]: See comment under controller.tf about only declaring these once.

**Comment by weitenghuang** [comment]
:shipit:

**Pull request merged**